### ISSUE TITLE
[Bug]: the default value is not used for mandatory fields

### ISSUE THREAD

**Issue by youwe-petervanderwal** [pimcore/pimcore#15311]
### Expected behavior

The default value is set on the (newly created) DataObject

### Actual behavior

Pimcore\Model\Element\ValidationException: Empty mandatory field [ fieldName ]

### Steps to reproduce

With (currently latest versions) Pimcore v10.5.20, DataHub v1.5.6 and DataImporter v1.7.5

1. Create a class definition, with
  a. Mark one field as mandatory field and also give it a default value (e.g. a status dropdown with default value "New")
2. Create a data importer definition, with
  a. Resolver configuration with a valid "Element Creation" config
  b. Resolver configuration publish strategy set to "Always publish" or "Publish new"
  c. Valid mapping configuration so that it can create new DataObjects (mainly, the system key should be set)
  d. Important: don't map the field created in step 1a.
3. Run an import

**Comment by youwe-petervanderwal** [comment]
What I've found so far, in Pimcore itself (so I actually don't know whether we should resolve this in Pimcore or in the DataImporter):

- `\Pimcore\Model\DataObject\Concrete::update()` first checks the validity of the DataObject (which will fail)
- After checking validity, it will call `$this->getDao()->update()`
- The `\Pimcore\Model\DataObject\Concrete\Dao::update()` will call `$fd->getDataForResource()` 
- Applicable field definitions will call `$data = $this->handleDefaultValue()`

For normal object creations it works because when you add a new object it will be unpublished initially (so the validity check will be skipped) and stored to the database (where the default value will be injected).

For the same reason (validity check not complaining), on non-mandatory fields the default value will also be set correctly.

**Comment by mcop1** [comment]
Hello @youwe-petervanderwal ,
just tried to reproduce this problem, but I couldn´t. 

Added a pretty basic input field to my data object, set it to mandatory and provided a default value: 
![grafik](https://github.com/pimcore/data-importer/assets/89011527/a55043d1-f747-4c84-89ba-e70721b220ba)

Created a new data-importer configuration and added a simple mapping for `SYSTEM KEY`, nothing else.
But import worked as expected and created a new object including the default value:
![grafik](https://github.com/pimcore/data-importer/assets/89011527/4d405e7d-214a-4d33-b6fd-4f6d80e6c7f2)
![grafik](https://github.com/pimcore/data-importer/assets/89011527/8c969ffd-ed92-4945-9b95-05894ae11ecb)

Did I miss something? Could you please try to reproduce the issue in the public demo?

**Comment by youwe-petervanderwal** [comment]
Hi @mcop1 

Looking at the last screenshot, the object isn't published (and therefore validation didn't took place).

This setting is relevant, I see I missed that in the "steps to reproduce" section, so I just updated it (default value is "No change, unpublish new").
![image](https://github.com/pimcore/data-importer/assets/60703382/867b52d3-7009-41a1-815a-1f7e447d86ca)

At https://demo.pimcore.com/admin/ :
![image](https://github.com/pimcore/data-importer/assets/60703382/167bd6df-0a99-45cf-a1f5-a434abead71c)

**Comment by mcop1** [comment]
Hi @youwe-petervanderwal ,

thanks, I missed the obvious, I guess. Can reproduce the problem, but I think it should be fixed in the core, a simple script like 
```
        $obj = new DataObject\Mandatory();
        $obj->setParentId(1);
        $obj->setPublished(true);
        $obj->setKey(uniqid("test-mandatory-action-"));
        $obj->save();
```

fails too (of course). 

I will move this issue to pimcore/pimcore.

**Comment by fashxp** [comment]
related to https://github.com/pimcore/pimcore/pull/5594?

**Comment by youwe-petervanderwal** [comment]
> related to #5594?

Indeed relates to, but back then the choice was made to set the default value in the DataObject (in PHP memory) at the moment it is going to be stored to the database (`getDataForResource`) but actually it should be available earlier (see issue description and  comment from mcop1, https://github.com/pimcore/pimcore/issues/15311#issuecomment-1571623416 ).

Maybe it should be set in the constructor but I don't know what side effects that would have

**Comment by dvesh3** [comment]
Fixed by #15812
